supporter-product-data/src/main/scala/com/gu/lambdas/ProcessSupporterRatePlanItemLambda.scala

```diff
@@ -0,0 +1,132 @@
+package com.gu.lambdas
+
+import com.amazonaws.services.lambda.runtime.Context
+import com.gu.conf.ZuoraQuerierConfig
+import com.gu.lambdas.ProcessSupporterRatePlanItemLambda.getSupporterRatePlanItemProcessor
+import com.gu.lambdas.SupporterRatePlanItemCodec.codec
+import com.gu.model.StageConstructors
+import com.gu.model.sqs.SqsEvent
+import com.gu.monitoring.SafeLogger
+import com.gu.monitoring.SafeLogger.Sanitizer
+import com.gu.okhttp.RequestRunners.configurableFutureRunner
+import com.gu.services.{AlarmService, ConfigService, ZuoraSubscriptionService}
+import com.gu.supporterdata.model.Stage.{DEV, PROD, UAT}
+import com.gu.supporterdata.model.{ContributionAmount, Stage, SupporterRatePlanItem}
+import com.gu.supporterdata.services.SupporterDataDynamoService
+import com.typesafe.scalalogging.StrictLogging
+import io.circe.Codec
+import io.circe.generic.semiauto.deriveCodec
+import io.circe.parser._
+import io.circe.syntax.EncoderOps
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.{Await, Future}
+import scala.concurrent.duration.DurationInt
+import scala.util.{Failure, Success}
+
+class ProcessSupporterRatePlanItemLambda(supporterRatePlanItemProcessor: SupporterRatePlanItemProcessor)
+    extends Handler[SqsEvent, Unit] {
+
+  def this() = this(getSupporterRatePlanItemProcessor())
+
+  override protected def handlerFuture(input: SqsEvent, context: Context) = {
+    SafeLogger.info(s"Received ${input.Records.length} records from the queue")
+    Future
+      .sequence(input.Records.map { record =>
+        val subscription = decode[SupporterRatePlanItem](record.body)
+        subscription match {
+          case Right(item) => supporterRatePlanItemProcessor.processItem(item)
+          case _ =>
+            SafeLogger.warn(s"Couldn't decode a SupporterRatePlanItem with body: ${record.body}")
+            Future.successful(()) // This should never happen so I don't think it's worth alerting on
+        }
+      })
+      .map(_ => ())
+  }
+}
+
+object ProcessSupporterRatePlanItemLambda {
+  def getSupporterRatePlanItemProcessor() = {
+    val stage = StageConstructors.fromEnvironment
+    val dynamoService = SupporterDataDynamoService(stage)
+    val alarmService = AlarmService(stage)
+    val futureProcessor = ConfigService(stage).load.map { config =>
```

Comment:
it looks like we will load the config on every invocation of the lambda.  If we make getSupporterRatePlanItemProcessor a val instead, it will only do it once, on cold start/class loading.